### ISSUE TITLE
Pooling::Cls should use first real token (attention mask), not index 0

### ISSUE THREAD

**Issue by amgix-dev** [StarlightSearch/EmbedAnything#213]
### Problem

`Pooling::Cls` in `rust/src/embeddings/local/pooling.rs` always takes the hidden state at sequence index 0:

    fn cls(output: &ModelOutput) -> Result<PooledOutputType, anyhow::Error> {
        match output {
            ModelOutput::Tensor(tensor) => tensor
                .get_on_dim(1, 0)  // always position 0
                ...
        }
    }

For padded batches, CLS pooling should use the first **non-padding** token (via `attention_mask`), not a fixed index 0. This matters when sequences are left-padded.

`Pooling::LastToken` already has padding awareness in the same file; CLS does not.

### Expected behavior

CLS pooling should:

1. Take `attention_mask` as input (like mean pooling)
2. Use the index of the first real token per sequence
3. Match the behavior used by common ModernBERT embedding checkpoints (e.g. Granite, gte-modernbert)

### Impact

CLS-pooled models may produce incorrect embeddings when batching with padding, especially left padding.

### Suggested fix

Update `Pooling::Cls` to accept `attention_mask` and gather the first real token per row.

**Comment by mikemikimike** [comment]
Hi! I’d like to work on this issue if it is still available. I can update CLS pooling to use the attention mask (including both Tensor and ndarray paths) and add CPU unit tests covering left-padded batches while preserving the existing no-mask behavior. Please confirm and assign the issue before I start. Thanks!

**Comment by amgix-dev** [comment]
I'm not working on it. But I'm not a maintainer and I don't know if anyone else is, but it's probably okay for you try to tackle this.

**Comment by akshayballal95** [comment]
This is merged now. Closing
